test_plans/queue_constructors.asciidoc

```diff
@@ -0,0 +1,82 @@
+For following constructors use created `syclContext`, selected device or `cts_selector`, `cts_async_handler` and check that synchronous exception with the `errc::invalid` error code is thrown:
+
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector, const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
+
+Note: test with and without both properties `enable_profiling` and `in_order`.
+
+=== Check property `in_order`
+
+Use property `in_order` with following constructors also using selected device, corresponding context, `cts_selector`, `cts_async_handler` and check that resulting queue member functions `has_property<in_order>()` and `is_in_order()` return `true`:
+
+* `queue(const property_list& propList = {})`
+* `queue(const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const DeviceSelector& deviceSelector,
+               const property_list& propList = {})`
+* `queue(const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const device& syclDevice, const property_list& propList = {})`
+* `queue(const device& syclDevice, const async_handler& asyncHandler,
+               const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector, const property_list& propList = {})`
+* `queue(const context& syclContext, const DeviceSelector& deviceSelector,
+               const async_handler& asyncHandler, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice, const property_list& propList = {})`
+* `queue(const context& syclContext, const device& syclDevice,
+               const async_handler& asyncHandler,
+               const property_list& propList = {})`
+
+If `device::has(sycl::aspect::usm_host_allocations)` is `false` this part of test is skipped.
+Submit two `single_task` to queue - first with long loop and second that changes a `usm` variable with host allocation.
+In the first task check that after loop the variable is unchanged to verify that `in_order` property adds the requirement that the SYCL queue provides in-order semantics.
+
+=== Check constructors with both properties
+
+Use `propList = {in_order, enable_profiling}` with following constructors also using selected device, corresponding context, `cts_selector`, `cts_async_handler` and check that resulting queue member finctions `has_property<enable_profiling>()`, `has_property<in_order>()` and `is_in_order()` return `true`:
```

Comment:
The `enable_profiling` property is an optional feature.  The test needs to check if the selected device has `aspect::queue_profiling` before constructing a queue with this property.  If the selected device doesn't have this aspect, this test should be skipped.

Reply:
Added condition `device::has(sycl::aspect::queue_profiling)` for test case.